test/iterator/basic_iterator.cpp

```diff
@@ -239,6 +239,20 @@ struct proxy_wrapper {
 		return *this;
 	}
 
+	proxy_wrapper const& operator=(const T& t) const
+	noexcept(std::is_nothrow_copy_assignable<T>::value)
+	{
+		get() = t;
+		return *this;
+	}
+
+	proxy_wrapper const& operator=(T&& t) const
+	noexcept(std::is_nothrow_move_assignable<T>::value)
+	{
+		get() = stl2::move(t);
+		return *this;
+	}
+
```

Comment:
I got strange ambiguities when I removed the non-const overloads, but didn't bother to track it down. Maybe it has something to do with the presence of the implicit conversion operator? I dunno. I was lazy.